components/ledger/internal/services/command/create_balance_transaction_operations_async.go

```
@@ -307,6 +310,21 @@ func (uc *UseCase) SendTransactionToRedisQueue(ctx context.Context, organization
 	logger, _, reqId, _ := libCommons.NewTrackingFromContext(ctx)
 	transactionKey := utils.TransactionInternalKey(organizationID, ledgerID, transactionID.String())
 
+	// Scope protection: a transaction that targets any internal-scope
+	// balance (e.g. auto-created overdraft reserves) MUST be rejected
+	// BEFORE the transaction is published to the Redis queue. This keeps
+	// system-managed balances out of the user-initiated mutation path
+	// across every entry point (HTTP, gRPC, DSL).
+	for _, b := range balances {
+		if b != nil && b.Settings != nil && b.Settings.BalanceScope == mmodel.BalanceScopeInternal {
+			err := pkg.ValidateBusinessError(constant.ErrDirectOperationOnInternalBalance, constant.EntityBalance, b.Alias)
+			logger.Log(ctx, libLog.LevelWarn, "Rejected transaction targeting internal balance",
```

golangci-lint (2.11.4)

[error] missing whitespace above this line (no shared variables above expr)

(wsl_v5)